src/borg/testsuite/__init__.py

```diff
@@ -54,6 +55,86 @@
     st_mtime_ns_round = -4  # only >1 microsecond resolution here?
 
 
+_symlinks_supported = None
+_fifos_supported = None
+_hardlinks_supported = None
+_utime_supported = None
+
+
+def generate_tempfile_path():
+    return os.path.join(tempfile.gettempdir(), 'borg-test-' + uuid.uuid4().hex)
+
+
+def are_symlinks_supported():
```

Comment:
[`@functools.lru_cache()`](file:///usr/share/doc/python/html/library/functools.html) allows you to avoid both the global variables and checking/setting them.